### DISCUSSION TITLE
TFT_eSPI_ext problems running demos

### DISCUSSION THREAD

**Discussion by smkr9500** [Bodmer/TFT_eSPI#2265] [Q&A - General] [answered]
Hi

I have tried to run the example sketch: font_test.ino. 
I get this error everytime I try:

E:\Dokumenter\Arduino\libraries\TFT_eSPI_ext-main\examples\TFT_and_sprite_demo\TFT_and_sprite_demo.ino:23:34: error: no matching function for call to 'TFT_eSPI_ext::TFT_eSPI_ext()'
 TFT_eSPI_ext etft = TFT_eSPI_ext();
                                  ^
In file included from E:\Dokumenter\Arduino\libraries\TFT_eSPI_ext-main\examples\TFT_and_sprite_demo\TFT_and_sprite_demo.ino:5:
e:\Dokumenter\Arduino\libraries\TFT_eSPI_ext-main/TFT_eSPI_ext.h:36:3: note: candidate: 'TFT_eSPI_ext::TFT_eSPI_ext(TFT_eSPI*)'
   TFT_eSPI_ext(TFT_eSPI *tft)
   ^~~~~~~~~~~~
e:\Dokumenter\Arduino\libraries\TFT_eSPI_ext-main/TFT_eSPI_ext.h:36:3: note:   candidate expects 1 argument, 0 provided
e:\Dokumenter\Arduino\libraries\TFT_eSPI_ext-main/TFT_eSPI_ext.h:29:7: note: candidate: 'TFT_eSPI_ext::TFT_eSPI_ext(const TFT_eSPI_ext&)'
 class TFT_eSPI_ext : public TFT_eSPI
       ^~~~~~~~~~~~
e:\Dokumenter\Arduino\libraries\TFT_eSPI_ext-main/TFT_eSPI_ext.h:29:7: note:   candidate expects 1 argument, 0 provided

exit status 1

Compilation error: no matching function for call to 'TFT_eSPI_ext::TFT_eSPI_ext()'

Br Sten

**Comment by MYCAMEL222** [comment]
I am getting the same error. Trying to use the larger fonts in the _ext library.

**Reply by smkr9500** [reply]
The thing is, I don't even know if need to use this example!! 
I need to have the danish character set with æøåÆØÅ.  
Both **U8g2_for_TFT_eSPI.h** and the **GFXFF** in Local_Custom_Fonts sketch can show the characters.

**Comment by smkr9500** [comment]
I think I found a solution. In both the font_test.ino and TFT_sprite_demo.ino.
Changing the line: **TFT_eSPI_ext etft = TFT_eSPI_ext();**  to **TFT_eSPI_ext etft = TFT_eSPI_ext(&tft);**
then both sketches run.

**Reply by MYCAMEL222** [reply]
I'll give that a try. Font frustration forced me to use the Smooth font option and I actually had to read Bodmer's instructions on how to make new fonts. I'll be doing that from now on. I tried using the Openfontrender library. I could get the ofr demos to run, but when I tried to use it in my sketch, I could never get the fonts to load.

The smooth font option is just fine and I'm storing the new fonts in array.h files. So far, works great.

**Comment by Bodmer** [comment, marked as answer]
Line 22 of the example should read:

TFT_eSPI_ext etft = TFT_eSPI_ext(&tft);

I will post a pull request.

**Comment by Bodmer** [comment]
Pull request issued. In meantime library fork here has the update:

https://github.com/Bodmer/TFT_eSPI_ext

**Reply by Sinamajidi** [reply]
Does that work well with PowerPoint_to_sketch(https://github.com/Bodmer/PowerPoint_to_sketch)?

I have used thst but aftet building generated code i got errors about exactly the thing you polled and after solving it, the paltformio esp32 compiler couldn't resolve setTTFFont in main program despite the existence of headers and the function set in the headers under that class, even after cleaning and rebuilding.
